adam_core/coordinates/covariances.py

```diff
@@ -232,10 +225,124 @@ def sample_covariance(
     -------
     samples : `~numpy.ndarray` (num_samples, D)
         The drawn samples row-by-row.
+    W: `~numpy.ndarray` (num_samples)
+        Weights of the samples.
+    W_cov: `~numpy.ndarray` (num_samples)
+        Weights of the samples to reconstruct covariance matrix.
     """
     normal = multivariate_normal(mean=mean, cov=cov, allow_singular=True)
     samples = normal.rvs(num_samples)
-    return samples
+    W = np.full(num_samples, 1 / num_samples)
+    W_cov = np.full(num_samples, 1 / num_samples)
+    return samples, W, W_cov
+
+
+def sample_covariance_sigma_points(
+    mean: np.ndarray, cov: np.ndarray, alpha: float = 1, kappa: float = 0.0
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Create sigma-point samples of a multivariate Gaussian distribution
+    with given mean and covariances.
+
+    Parameters
+    ----------
+    mean : `~numpy.ndarray` (D)
+        Multivariate mean of the Gaussian distribution.
+    cov : `~numpy.ndarray` (D, D)
+        Multivariate variance-covariance matrix of the Gaussian distribution.
+    alpha : float, optional
+        Spread of the sigma points between 1e^-2 and 1.
+    kappa : float, optional
+        Secondary scaling parameter usually set to 0.
+
+    Returns
+    -------
+    sigma_points : `~numpy.ndarray` (2 * D + 1, D)
+        Sigma points drawn from the distribution.
+    W: `~numpy.ndarray` (2 * D + 1)
+        Weights of the sigma points.
+    W_cov: `~numpy.ndarray` (2 * D + 1)
+        Weights of the sigma points to reconstruct covariance matrix.
+
+    References
+    ----------
+    [1] Wan, E. A; Van Der Merwe, R. (2000). The unscented Kalman filter for nonlinear estimation.
+        Proceedings of the IEEE 2000 Adaptive Systems for Signal Processing,
+        Communications, and Control Symposium, 153-158.
+        https://doi.org/10.1109/ASSPCC.2000.882463
+    """
+    # Calculate the dimensionality of the distribution
+    D = mean.shape[0]
+
+    # See equation 15 in Wan & Van Der Merwe (2000) [1]
+    N = 2 * D + 1
+    sigma_points = np.empty((N, D))
+    W = np.empty(N)
+    W_cov = np.empty(N)
+
+    # Calculate the scaling parameter lambda
+    lambd = alpha**2 * (D + kappa) - D
```

Comment:
This is going to be a negative value. 

[Julier & Uhlmann 1997](https://www.cs.unc.edu/~welch/kalman/media/pdf/Julier1997_SPIE_KF.pdf) warns:
<img width="798" alt="image" src="https://github.com/B612-Asteroid-Institute/adam_core/assets/880413/497b2723-6c01-4ef1-8c64-78100d51193a">

(they call your lambda "κ" and don't decompose it into alpha and kappa, but I think its the same thing).

I can't tell whether this is an issue, but it seems worth mentioning.